### DISCUSSION TITLE
How to mock Graphql Endpoint

### DISCUSSION THREAD

**Discussion by DarkBitz** [urql-graphql/urql#1993] [Q&A] [answered]
Hello, I would like to know the easiest way for you to mock the client while developing? As a React Frontend Developer, I don't always want to have one Docker container running for the database and another one for the backend. How do you solve the problem?

**Comment by JoviDeCroock** [comment, marked as answer]
You can replace the ´fetchExchange´ with the `executeExchange` as a way to mock responses https://github.com/FormidableLabs/urql/tree/main/exchanges/execute

**Comment by taneba** [comment]
Or you can use msw to mock request instead of mocking client.
https://mswjs.io/
